File: server/service/contest_service/contest.go
package contest_service

import "github.com/zilanlann/acmer-manage-system/server/model"

type ContestList struct {
	Contests []model.Contest `json:"list"`
	Total    int             `json:"total"`
}

func (c *ContestList) Get() (err error) {
	c.Contests, err = model.GetAllContestList()
	c.Total = len(c.Contests)
	return
}

type Contest struct {
	model.Contest
}

func (c *Contest) Get() (err error) {
	c.Contest, err = model.GetContestInfo(c.ID)
	return
}

func (c *Contest) Create() (err error) {
	err = model.CreateContest(&c.Contest)
	return
}

func (c *Contest) Update() (err error) {
	err = model.UpdateContest(&c.Contest)
	return
}

func (c *Contest) Delete() (err error) {
	err = model.DeleteContest(c.ID)
	return
}

type Team struct {
	model.Team
}

func (t *Team) Create() (err error) {
	err = model.CreateTeam(&t.Team)
	return
}

func (t *Team) Update() (err error) {
	err = model.UpdateTeam(&t.Team)
	return
}

func (t *Team) Delete() (err error) {
	err = model.DeleteTeam(t.ID)
	return
}

type Contestant struct {
	model.Contestant
}

func (c *Contestant) Create() (err error) {
	err = model.CreateContestant(&c.Contestant)
	return
}

func (c *Contestant) Update() (err error) {
	err = model.UpdateContestant(&c.Contestant)
	return
}

func (c *Contestant) Delete() (err error) {
	err = model.DeleteContestant(c.ID)
	return
}

File: web/src/api/contest.ts
import { http } from "@/utils/http";
import { baseUrlApi, v1BaseUrlApi } from "./utils";

type Result = {
  success: boolean;
  code: number;
  message: string;
  data?: Array<any>;
};

type ResultTable = {
  success: boolean;
  data?: {
    /** 列表数据 */
    list: Array<any>;
    /** 总条目数 */
    total?: number;
    /** 每页显示条目个数 */
    pageSize?: number;
    /** 当前页数 */
    currentPage?: number;
  };
};

/** 获取比赛管理列表 */
export const getContestList = () => {
  return http.request<ResultTable>("get", v1BaseUrlApi("contests"));
};

/** 添加比赛 */
export const createContest = (data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("contest"), { data });
};

/** 更新比赛 */
export const updateContest = (id?: number, data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("contest/" + id), { data });
};

/** 删除比赛 */
export const deleteContest = (id?: number) => {
  return http.request<Result>("delete", v1BaseUrlApi("contest/") + id);
};

/** 添加队伍 */
export const createTeam = (data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("team"), { data });
};

/** 修改队伍 */
export const updateTeam = (id?: number, data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("team/" + id), { data });
};

/** 删除队伍 */
export const deleteTeam = (id?: number, data?: object) => {
  return http.request<Result>("delete", v1BaseUrlApi("team/" + id), { data });
};

/** 添加选手 */
export const createContestant = (data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("contestant"), { data });
};

/** 修改选手 */
export const updateContestant = (id?: number, data?: object) => {
  return http.request<Result>("put", v1BaseUrlApi("contestant/" + id), {
    data
  });
};

/** 删除选手 */
export const deleteContestant = (id?: number, data?: object) => {
  return http.request<Result>("delete", v1BaseUrlApi("contestant/" + id), {
    data
  });
};
